### PR TITLE
SMS workflow actions only for Teams

### LINKED ISSUE #5916: [CAL-556] SMS reminders only for paying customers
### Is your proposal related to a problem?

From my perspective, an SMS reminder should only be for paying customers because SMS cost money.

<sub>[CAL-556](https://linear.app/calcom/issue/CAL-556/sms-reminders-only-for-paying-customers)</sub>

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Makes SMS workflow actions only available for paying users (users that belong to a team). For all other users, the SMS actions are disabled in the dropdown. 

Fixes #5916

Loom Video: https://www.loom.com/share/52e13c89ebd2441a9b9ca7519c0d4c5a

**Environment**: Staging(main branch)

### CODE REVIEW

File: packages/trpc/server/routers/viewer/workflows.tsx [hunk 1]
```diff
@@ -519,6 +530,14 @@ export const workflowsRouter = router({
           });
           //step was edited
         } else if (JSON.stringify(oldStep) !== JSON.stringify(newStep)) {
+          if (
+            !userWorkflow.user.teams.length &&
+            !isSMSAction(oldStep.action) &&
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows.tsx:535]: We still allow to edited already existing SMS workflow actions for users that don't belong to a team

File: packages/trpc/server/routers/viewer/workflows.tsx [hunk 2]
```diff
@@ -33,6 +33,12 @@ import { TRPCError } from "@trpc/server";
 
 import { router, authedProcedure } from "../../trpc";
 
+function isSMSAction(action: WorkflowActions) {
+  if (action === WorkflowActions.SMS_ATTENDEE || action === WorkflowActions.SMS_NUMBER) {
+    return true;
+  }
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/workflows.tsx:39]:
```suggestion
  return action === WorkflowActions.SMS_ATTENDEE || action === WorkflowActions.SMS_NUMBER;
```

File: packages/trpc/server/routers/viewer.tsx [hunk 3]
```diff
@@ -1142,6 +1142,17 @@ const loggedInViewerRouter = router({
...
+  isTeamsPlan: authedProcedure.query(async ({ ctx }) => {
+    const user = await ctx.prisma.user.findFirst({
+      where: {
+        id: ctx.user.id,
+      },
+      select: {
+        teams: true,
+      },
+    });
+    return (user && user.teams.length > 0) as boolean;
```
Comment (zomars) [inline packages/trpc/server/routers/viewer.tsx:1154]:
```suggestion
    const membershipCount = await ctx.prisma.membership.count({
      where: { userId: ctx.user.id },
    });
    return membershipCount > 0;
```

We only need to count memberships

File: packages/features/ee/workflows/components/WorkflowStepContainer.tsx [hunk 4]
```diff
@@ -93,7 +94,7 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
     form.getValues("trigger") === WorkflowTriggerEvents.AFTER_EVENT
   );
 
-  const actionOptions = getWorkflowActionOptions(t);
+  const actionOptions = getWorkflowActionOptions(t, isTeamsPlan);
```
Comment (zomars) [inline packages/features/ee/workflows/components/WorkflowStepContainer.tsx:97]: How about instead of prop-drilling we get this from the backend directly?

**Review by zomars — CHANGES_REQUESTED** [review]
Nice work @CarinaWolli. Blocking to make some adjustments. 🙏🏽

**Review by zomars — APPROVED** [review]
Ship it

**Pull request merged**